iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNodeServerCommandLine.java

```diff
@@ -135,57 +163,60 @@ private void doRemoveDataNode(String[] args)
-        dataNodeLocations =
-            client.getDataNodeConfiguration(-1).getDataNodeConfigurationMap().values().stream()
-                .map(TDataNodeConfiguration::getLocation)
-                .filter(location -> endPoints.contains(location.getClientRpcEndPoint()))
-                .collect(Collectors.toList());
-      } catch (TException | ClientManagerException e) {
-        LOGGER.error("Get data node locations failed", e);
-      }
-    } catch (BadNodeUrlException e) {
-      try (ConfigNodeClient client =
-          ConfigNodeClientManager.getInstance().borrowClient(ConfigNodeInfo.CONFIG_REGION_ID)) {
-        for (String id : args.split(",")) {
-          if (!isNumeric(id)) {
-            LOGGER.warn("Incorrect id format {}, skipped...", id);
-            continue;
-          }
-          List<TDataNodeLocation> nodeLocationResult =
-              client
-                  .getDataNodeConfiguration(Integer.parseInt(id))
-                  .getDataNodeConfigurationMap()
-                  .values()
-                  .stream()
-                  .map(TDataNodeConfiguration::getLocation)
-                  .collect(Collectors.toList());
-          if (nodeLocationResult.isEmpty()) {
-            LOGGER.warn("DataNode {} is not in cluster, skipped...", id);
-            continue;
-          }
-          if (!dataNodeLocations.contains(nodeLocationResult.get(0))) {
-            dataNodeLocations.add(nodeLocationResult.get(0));
-          }
-        }
-      } catch (TException | ClientManagerException e1) {
-        LOGGER.error("Get data node locations failed", e);
-      }
+    List<NodeCoordinate> nodeCoordinates = parseCoordinates(args);
+    try (ConfigNodeClient client =
+        configNodeClientManager.borrowClient(ConfigNodeInfo.CONFIG_REGION_ID)) {
+      dataNodeLocations =
+          client.getDataNodeConfiguration(-1).getDataNodeConfigurationMap().values().stream()
+              .map(TDataNodeConfiguration::getLocation)
+              .filter(
+                  location ->
+                      nodeCoordinates.stream()
+                          .anyMatch(nodeCoordinate -> nodeCoordinate.matches(location)))
+              .collect(Collectors.toList());
+    } catch (TException | ClientManagerException e) {
+      LOGGER.error("Get data node locations failed", e);
     }
+
     return dataNodeLocations;
   }
+
+  protected List<NodeCoordinate> parseCoordinates(String coordinatesString) {
+    // Multiple nodeIds are separated by ","
+    String[] nodeIdStrings = coordinatesString.split(",");
+    List<NodeCoordinate> nodeIdCoordinates = new ArrayList<>(nodeIdStrings.length);
+    for (String nodeId : nodeIdStrings) {
+      // In the other case, we expect it to be a numeric value referring to the node-id
+      if (NumberUtils.isCreatable(nodeId)) {
+        nodeIdCoordinates.add(new NodeCoordinateNodeId(Integer.parseInt(nodeId)));
+      }
```

Comment:
maybe we can log something to remind user here?

Reply:
Done